### DISCUSSION TITLE
How do I use upstream?

### DISCUSSION THREAD

**Discussion by bkeenke** [NginxProxyManager/nginx-proxy-manager#3521] [Q&A]
There is a config that uses the ngx_stream_module.so module
I add this config to folder modules-available 

> upstream.conf

```
map_hash_bucket_size 128;
map $ssl_preread_server_name $sni_name {
   domain.com server1;
}
server {
   listen 443 reuseport;
   proxy_pass $sni_name;
   proxy_protocol on;
   ssl_preread on;
}
upstream server1 {
   server 127.0.0.1:4443;
}
```
how do I use it in Nginx Proxy Manager?

**Comment by dbryuhanov542** [comment]
Check [this](https://nginxproxymanager.com/advanced-config/#custom-nginx-configurations)

**Comment by PastaGringo** [comment]
Did you find out? 👍
